Snake/src/snakeBoard.js

```diff
@@ -0,0 +1,446 @@
+        }
+
+        if (snakeHead.x === snakeBoard.width || snakeHead.x === (snakeBoard.width - this.squareSize / 2)) {
+            snakeHead.x = 0;
+        }
+
+        if (snakeHead.y < 0) {
+            snakeHead.y = snakeBoard.height - this.squareSize;
+        }
+
+        if (snakeHead.y === snakeBoard.height || snakeHead.y === (snakeBoard.height - this.squareSize / 2)) {
+            snakeHead.y = 0;
+        }
+
+        newSnakePos.unshift(snakeHead);
+        newSnakePos.pop();
+
+        this.setState({
+            snake: newSnakePos
+        });
+    }
+
+    moveSnakeWithBorders() {
+        let newSnakePos = [...this.state.snake];
+        const head = { x: this.state.snake[0].x + this.horizontalAdjustment, y: this.state.snake[0].y + this.verticalAdjustment };
+
+        newSnakePos.unshift(head);
+        newSnakePos.pop();
+
+        this.setState({ snake: newSnakePos });
+    }
+
+    setLastKeyPressed(keyCode) {
+        this.setState({
+            lastKeyPressed: keyCode
+        });
+    }
+
+    changeDirection(event) {
+        const keyLeft = 37;
+        const keyUp = 38;
+        const keyRight = 39;
+        const keyDown = 40;
+        const keyPressed = event.keyCode;
+        const { lastKeyPressed } = this.state;
+
+        if (keyPressed === keyLeft && lastKeyPressed !== keyRight) {
+            this.horizontalAdjustment = -this.squareSize;
+            this.verticalAdjustment = 0;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+
+        if (keyPressed === keyUp && lastKeyPressed !== keyDown) {
+            this.horizontalAdjustment = 0;
+            this.verticalAdjustment = -this.squareSize;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+
+        if (keyPressed === keyRight && lastKeyPressed !== keyLeft) {
+            this.horizontalAdjustment = this.squareSize;
+            this.verticalAdjustment = 0;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+
+        if (keyPressed === keyDown && lastKeyPressed !== keyUp) {
+            this.horizontalAdjustment = 0;
+            this.verticalAdjustment = this.squareSize;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+    }
+
+    getRandomPosition(board) {
+        const x = Math.round(Math.random() * board.current.width);
+        const y = Math.round(Math.random() * board.current.height);
+
+        return {
+            x: x - x % this.squareSize + (this.squareSize / 2),
+            y: y - y % this.squareSize + (this.squareSize / 2)
+        };
+    }
+
+    setFoodPosition(foodPos) {
+        this.setState({ food: foodPos });
+    }
+
+    drawFood() {
+        const snakeBoardStyles = this.getSnakeBoardStyles();
+        const radius = this.props.snakeSize / 2;
+        let { x: foodX, y: foodY } = this.state.food;
+
+        snakeBoardStyles.beginPath();
+        snakeBoardStyles.arc(foodX, foodY, radius, 0, 2 * Math.PI, false);
+        snakeBoardStyles.fillStyle = 'lightgreen';
```

Comment:
Colors should be in consts on top of the file or have a file which stores Color consts